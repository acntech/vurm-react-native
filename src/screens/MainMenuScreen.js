import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  SafeAreaView,
} from "react-native";
import { GAME_TITLE } from "../constants";

export const startGameLabel = "Start Game";
export const instructionsLabel = "Instructions";

const HomeScreen = ({ navigation }) => {
  const buttonLabels = [startGameLabel, instructionsLabel];
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.label}>{GAME_TITLE}</Text>
      <View style={styles.column}>
        {buttonLabels.map((buttonLabel) => (
          <TouchableOpacity
            key={buttonLabel}
            onPress={() => navigation.navigate(buttonLabel)}
            style={[
              styles.button,
              buttonLabel === startGameLabel && styles.primaryButton,
            ]}
          >
            <Text
              style={[
                styles.buttonLabel,
                buttonLabel === startGameLabel && styles.primaryButtonLabel,
              ]}
            >
              {buttonLabel}
            </Text>
          </TouchableOpacity>
        ))}
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
  },
  column: {
    flexDirection: "column",
    alignContent: "center",
  },
  button: {
    paddingHorizontal: 8,
    paddingVertical: 8,
    borderRadius: 4,
    backgroundColor: "oldlace",
    alignSelf: "center",
    marginBottom: 5,
    minWidth: "50%",
  },
  primaryButton: {
    backgroundColor: "coral",
    borderWidth: 0,
  },
  buttonLabel: {
    fontSize: 16,
    fontWeight: "400",
    color: "coral",
    textAlign: "center",
  },
  primaryButtonLabel: {
    color: "white",
  },
  label: {
    textAlign: "center",
    marginBottom: 10,
    fontSize: 26,
  },
});

export default HomeScreen;
