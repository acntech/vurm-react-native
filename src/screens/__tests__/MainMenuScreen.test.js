import React from "react";
import { render } from "@testing-library/react-native";
import MainMenuScreen from "../MainMenuScreen";

describe("<MainMenuScreen />", () => {
  it("has 2 children", () => {
    // Arrange & Act
    const tree = render(<MainMenuScreen />).toJSON();

    // Assert
    expect(tree.children.length).toBe(2);
  });
  it("renders correctly", () => {
    // Arrange & Act
    const tree = render(<MainMenuScreen />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
  });
});
