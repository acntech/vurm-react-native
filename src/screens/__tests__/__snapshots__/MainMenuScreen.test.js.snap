// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenuScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 26,
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Vurm
  </Text>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "column",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "coral",
          "borderRadius": 4,
          "borderWidth": 0,
          "marginBottom": 5,
          "minWidth": "50%",
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "textAlign": "center",
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        Start Game
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "oldlace",
          "borderRadius": 4,
          "borderWidth": 0,
          "marginBottom": 5,
          "minWidth": "50%",
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "textAlign": "center",
            },
            Object {
              "color": "coral",
            },
          ]
        }
      >
        Instructions
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "oldlace",
          "borderRadius": 4,
          "borderWidth": 0,
          "marginBottom": 5,
          "minWidth": "50%",
          "opacity": 1,
          "paddingHorizontal": 8,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "400",
              "textAlign": "center",
            },
            Object {
              "color": "coral",
            },
          ]
        }
      >
        Leaderboard
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
